src/sotoki/utils/preparation.py

```diff
@@ -16,11 +16,31 @@
 
 from .shared import logger
 from .misc import has_binary, get_available_memory
-from ..constants import UTF8
+from ..constants import UTF8, UTF16LE
 
 has_gnusort = has_binary("sort")
 
 
+def reencode_file(src: pathlib.Path):
+    """Reencode a file from dump format (UTF-16-LE as of March 2024) to expected format (UTF8)
+
+    This is based on a streaming on-the-fly reencoding of file chunks to limit memory pressure.
+
+    During few instants, there will be two versions of the same content on the filesystem, one
+    in previous encoding and one in target encoding, filesystem needs enough space for that.
+    """
+    BLOCKSIZE = 1048576
```

Comment:
I am worried that this naive implementation will raise decoding or encoding errors when reading a fixed number of bytes if the boundary is in the middle of a character/grapheme.

Do you know what would happen? Is is possible that we raise a decode or encode error ? Is it possible that we alter the content by misinterpreting a grapheme? Is `open` magically handling this? I believe `codecs.open` does but I don't know if the builtin `open` does…

Reply:
Nice catch ! 

`codecs.open` does not seems to help, at least I achieved to make it fail with the best emoji possible for this test: 🤷🏻‍♂️

I've changed the code to read line by line ; probably a bit slower but negligible compared to the rest of the processing. And at least we are sure to not split in the middle of a grapheme.